src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -0,0 +1,428 @@
+    auto viewTf = rScene.get_registry().view<ACompTransform>();
+
+    // Apply transform changes after the Newton world(s) updates
+    for (ActiveEnt ent : viewBody)
+    {
+        auto &entNwtBody      = viewNwtBody.get<ACompNwtBody>(ent);
+        auto &entTransform    = viewTf.get<ACompTransform>(ent);
+
+        float mass, x, y, z;
+        NewtonBodyGetMass(entNwtBody.body(), &mass, &x, &y, &z);
+
+        // Get new transform matrix from newton
+        NewtonBodyGetMatrix(entNwtBody.body(), entTransform.m_transform.data());
+    }
+}
+
+void SysNewton::find_colliders_recurse(
+        ActiveScene& rScene, ActiveEnt ent, Matrix4 const &transform,
+        NewtonWorld const* nwtWorld, NewtonCollision *rCompound)
+{
+    ActiveReg_t &rReg = rScene.get_registry();
+    ActiveEnt nextChild = ent;
+
+    // iterate through siblings of ent
+    while(nextChild != entt::null)
+    {
+        auto const &childHeir = rScene.reg_get<ACompHierarchy>(nextChild);
+        auto const *childTransform = rScene.get_registry().try_get<ACompTransform>(nextChild);
+
+        if (childTransform != nullptr)
+        {
+            Matrix4 childMatrix = transform * childTransform->m_transform;
+
+            if (rReg.all_of<ACompSolidCollider>(nextChild))
+            {
+                auto* childNwtCollider = rReg.try_get<ACompNwtCollider>(nextChild);
+
+                NewtonCollision const *collision;
+
+                if (nullptr == childNwtCollider)
+                {
+                    // No Newton collider exists yet
+                    collision = NewtonCreateSphere(nwtWorld, 0.5f, 0, NULL);
+                    rReg.emplace<ACompNwtCollider>(nextChild, collision);
+                    ACompNwtCollider f(collision);
+                }
+                else
+                {
+                    // Already has Newton collider
+                    collision = childNwtCollider->collision();
+                }
+
+                // Set transform relative to root body
+                Matrix4 f = Matrix4::translation(childMatrix.translation());
+                NewtonCollisionSetMatrix(collision, f.data());
+
+                // Add body to compound collision
+                NewtonCompoundCollisionAddSubCollision(rCompound, collision);
+            }
+
+            find_colliders_recurse(rScene, childHeir.m_childFirst, childMatrix,
+                                   nwtWorld, rCompound);
+        }
+
+        nextChild = childHeir.m_siblingNext;
+    }
+}
+
+void SysNewton::create_body(ActiveScene& rScene, ActiveEnt ent,
+                            NewtonWorld const* nwtWorld)
+{
+    ActiveReg_t &rReg = rScene.get_registry();
+
+    auto const& entHier = rReg.get<ACompHierarchy>(ent);
+    auto const& entTransform = rReg.get<ACompTransform>(ent);
+    auto const* entShape = rReg.try_get<ACompShape>(ent);
+
+    if ((! rReg.all_of<ACompSolidCollider>(ent)) || (entShape == nullptr))
```

Comment:
if ( ! rReg.all_of<ACompSolidCollider>(ent) || (entShape == nullptr))

the ! doesn't have the same operator precedence issues that i'm normally concerned about